### DISCUSSION TITLE
What does draftsman-update.exe ?

### DISCUSSION THREAD

**Discussion by flameSla** [redruin1/factorio-draftsman#36] [Q&A] [answered]
I didn't find the source code for this file.
I have 5 different versions of Factorio installed.
How draftsman-update.exe finds a folder with mods?

**Comment by redruin1** [comment, marked as answer]
The source code for `draftsman-update` it can be found in the `update()` function in the `draftsman.env` module [here](https://github.com/redruin1/factorio-draftsman/blob/4eca15ceede2745e407bfccf5860ae413a8e4372/draftsman/env.py#L1751).

The mods you want to work with have to be copied over into the source folder of wherever Draftsman was installed; it doesn't seek out a mod folder in the filesystem. ([See this section for more information.](https://factorio-draftsman.readthedocs.io/en/latest/handbook/modding.html#:~:text=Next%2C%20determine%20where%20your%20Draftsman%20installation%20is%20located)) Draftsman has it's own separate mod folder called `factorio-mods`; find where Draftsman was installed, copy the mods that you want it to run with to that folder, and then run `draftsman-update` to update the data with the newly added mods.

I suppose Draftsman *could* search for a mods folder, though that sounds like a nightmare to test, in addition to the system-specific differences in installation structure. Keeping it as a separate folder also ensures a degree of separation from the mods you are playing Factorio with and ones you're developing with; if you download someone else's script that expects a certain set of mods, do you really want to install those mods into your main game just for a single script?

**Reply by flameSla** [reply]
Copying a file to a folder is inconvenient. I have 1 GB of mods. I don't want to constantly copy 1GB from different folders.

Can do so?

from draftsman.env import update
update(verbose=True, show_logs=True, mods_path = 'Some super-duper folder\Factorio\mods' )

What is the threat of a constant call to the "update()" function in the program? Only by slowing down the program?

**Reply by redruin1** [reply]
Does the 1GB have to constantly be copied? The way it works right now you only have to copy the mods to that folder once, and it should work until you need to change the mods again, unless the mods themselves are constantly changing. What exactly are you trying to do?

Moving around 1GB (thats a **LOT** of mods!) I can understand to be inconvenient, however. If that's truly necessary and something you'd like to avoid, it shouldn't be very hard to add a `path="..."` option. Create an issue and I'll see if I can put it in.

Constantly calling `update()` shouldn't cause any problems (as far as I know) aside from probably being a little slow, as it's basically akin to relaunching Factorio every time you run `draftsman-update`.

**Reply by redruin1** [reply]
As of version `1.0.2`, you can now specify a `--path` argument to `draftsman-update` so you can point to an external folder to read mods from.
